File: Algorithms/Kotlin/BubbleSort.kt
fun Bubble_Sort(numbers: Array<Int>) {
    val N:Int = numbers.size
    for (pass in 0 until (N - 1)) {
        // A single pass of bubble sort
        for (currentPosition in 0 until (N - pass - 1)) {
            // This is a single step
            if (numbers[currentPosition] > numbers[currentPosition + 1]) {
                val tmp = numbers[currentPosition]
                numbers[currentPosition] = numbers[currentPosition + 1]
                numbers[currentPosition + 1] = tmp
            }
        }
    }

    //Printing Sorted Array
    for (i in 0 until (N-1)) {
        print(numbers[i])
    }
}

fun main(args: Array<String>){
    var numbers = Array<Int>(10){0}
    val N:Int = numbers.size

    for (i in 0 until (N - 1)){
        numbers[i] = readLine()!!.toInt()
    }

    Bubble_Sort(numbers)
}

File: Algorithms/C/Bubblesort.c
#include <stdio.h>

/* Sorting of an array using bubble sort */

int main()
{
    int n,i,j,k,l;
    printf("Enter number of elements to be sorted: ");
    scanf("%d",&n);
    int arr[n];
    printf("Enter numbers: ");
    for(i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }

    for(j=0;j<n-1;j++){
        for(k=0;k<n-j-1;k++){
            if(arr[k]>arr[k+1]){
                arr[k]=arr[k]+arr[k+1];
                arr[k+1]=arr[k]-arr[k+1];
                arr[k]=arr[k]-arr[k+1];
            }
        }
   }

    printf("The sorted array is: ");
    for(l=0;l<n;l++)
    printf("%d\n",arr[l]);
}
